src/backend/wsServer.ts

```diff
@@ -75,15 +79,23 @@ function hasPermission(playerId: PlayerId, gameId: GameId, gameEvent: GameEvent)
 		// to power to increase or decrease scores is the server
 		// so we may change this later
 		return playerId === gameEvent.data.id;
+	} else if (gameEvent.type === 'change-player-name') {
+		// players can only change their own names
+		return playerId === gameEvent.data.id;
 	}
-
 	return true;
 }
 
+/**
```

Comment:
if you're using this pr to learn how to create a new game event the rest of this file can be ignored, the changes are unrelated to implementing the ChangePlayerNameEvent.